### DISCUSSION TITLE
Service principal authentication between Logic app and Azure Blob storage

### DISCUSSION THREAD

**Discussion by Diviyan-S** [Azure/logicapps#994] [Help Needed] [answered]
Hello,

I have been trying to create API connection between logic app consumption and blob storage. In the portal, could create API connection by selecting "Service principal authentication" along with Tenant id, Client id, Client Secret. All good and API created, could create the blob as well.

Then started to prepare Logic app ARM template for automation (to deploy from azure DevOps). I have followed Microsoft doc https://learn.microsoft.com/en-us/azure/logic-apps/logic-apps-azure-resource-manager-templates-overview. 

Configured ARM template Microsoft.Web/connections as below. 

![image](https://github.com/Azure/logicapps/assets/123982879/a5468e19-9919-4bf3-bd8e-3e4f0a470e06)

Defined the parameters as below:

![image](https://github.com/Azure/logicapps/assets/123982879/a0d03b9c-5882-43f6-9db0-a91728e2e3f4)

Parameter file prepared as below:

![image](https://github.com/Azure/logicapps/assets/123982879/4629f3ac-632f-4b03-a719-1fdc874c166e)

**When try to deploy these from azure DevOps getting below error in release.**

**errorCode: ParameterNotDefined. 
Message: Parameter  'token:clientId' is not allowed on the connection since it was not defined as a connection parameter when the API was registered.**

**Am I missing any other configurations?
Can create service principal authentication through ARM templates?
Any help/suggestions would be appreciated...**

**Comment by subhash1692** [comment]
I am facing the same issue, when trying to create a connection from the logic app page, I can do it but when I tried from the CICD deployment pipeline, I got the same error, How to solve this?

**Reply by Diviyan-S** [reply, marked as answer]
Hi,

I was managed to configure CICD for this...

Please find the below for web connection. Let me know if this solved your issue or not....

`{
	"type": "Microsoft.Web/connections",
	"apiVersion": "2016-06-01",
	"location": "[parameters('logicAppLocation')]",
	"name": "[parameters('ConnectionName')]",
	"properties": {
		"api": {
			"id": "[concat('/subscriptions/',subscription().subscriptionId,'/providers/Microsoft.Web/locations/',parameters('logicAppLocation'),'/managedApis/azureblob')]"
		},
		"displayName": "[parameters('ConnectionName')]",
		"parameterValueSet": {
			"name": "servicePrincipalAuth",
			"values": {
				"token:clientId": {
					"value": "[parameters('ClientId')]"
				},
				"token:clientSecret": {
					"value": "[parameters('ClientSecret')]"
				},
				"token:TenantId": {
					"value": "[parameters('TenantID')]"
				}
			}
		}
	}
}`

For any discussion please connect over LinkedIn - https://www.linkedin.com/in/diviyan-s-49068a176.
